metadata/src/main/java/org/apache/kafka/metadata/migration/KRaftMigrationDriver.java

```diff
@@ -474,6 +472,7 @@ class MetadataChangeEvent extends MigrationEvent {
         private final MetadataImage image;
         private final MetadataProvenance provenance;
         private final boolean isSnapshot;
+        CompletableFuture<Void> future = new CompletableFuture<>();
```

Comment:
do we need this? it's expensive, and I can find 0 uses of it

Reply:
If we want to follow this pattern where the tests `KRaftMigrationDriverTest` are signalled when this event is processed, and at the same time avoid the cost of adding a future to every `SendRPCsToBrokersEvent`, `enqueueMetadataChangeEvent` could take a future in argument instead. Prod code would pass an empty/noop future. Alternatively the unit tests could loop until the assertions are verified or a timeout is reached.

Reply:
It's used to avoid having a bunch of wait-for conditions in the tests. Let me try your suggestion @Hangleton

Reply:
Thanks, looks good now.